src/bundle-manifest.ts

```diff
@@ -219,15 +216,85 @@ export function mergeBundles(bundles: Bundle[]): Bundle {
 /**
  * A simple function for generating shared bundle names based on a counter.
  */
-export function sharedBundleUrlMapper(bundles: Bundle[]): UrlString[] {
+export function sharedBundleUrlMapper(bundles: Bundle[]): Map<UrlString, Bundle> {
   let counter = 0;
-  const urls: UrlString[] = [];
+  const urlMap = new Map<UrlString, Bundle>();
   for (const bundle of bundles) {
     if (bundle.entrypoints.size === 1) {
-      urls.push(Array.from(bundle.entrypoints)[0]);
+      urlMap.set(Array.from(bundle.entrypoints)[0], bundle);
     } else {
-      urls.push(`shared_bundle_${++counter}.html`);
+      urlMap.set(`shared_bundle_${++counter}.html`, bundle);
+    }
+  }
+  return urlMap;
+}
+
+function setIntersection<T>(sets: Set<T>[]): Set<T> {
+  return sets.reduce((previous, current) => {
+    const reduced = new Set<T>();
+    for (let entry of previous) {
+      if (current.has(entry)) {
+        reduced.add(entry);
+      }
     }
+    return reduced;
+  });
+}
+
+function getEntrypointSets(bundles: Bundle[]): Set<string>[] {
+  const list: Set<string>[] = [];
+  for (let bundle of bundles) {
+    list.push(bundle.entrypoints);
+  }
+  return list;
+}
+
+/**
+ * Attempts to generate shared bundle names by identifying unique endpoints,
```

Comment:
You are using the word `endpoints` here instead of `entrypoints`.  Is this a mistake or does this mean something different?